addon/authenticators/oauth2-password-grant.js

```diff
@@ -434,15 +424,22 @@ export default BaseAuthenticator.extend({
     const serverTokenEndpoint = this.get('serverTokenEndpoint');
     return new RSVP.Promise((resolve, reject) => {
       this.makeRequest(serverTokenEndpoint, data).then((response) => {
-        run(() => {
-          expiresIn = response['expires_in'] || expiresIn;
-          refreshToken = response['refresh_token'] || refreshToken;
-          const expiresAt = this._absolutizeExpirationTime(expiresIn);
-          const data = assign(response, { 'expires_in': expiresIn, 'expires_at': expiresAt, 'refresh_token': refreshToken });
-          this._scheduleAccessTokenRefresh(expiresIn, null, refreshToken);
-          this.trigger('sessionDataUpdated', data);
-          resolve(data);
-        });
+        deprecatePlainReturnValuesFromMakeRequest(response);
+
+        const rejectWithResponse = this.get('rejectWithResponse');
+        deprecateNotRejectingWithResponse(rejectWithResponse);
+
+        this._parseResponse(response, rejectWithResponse).then((json) => {
+          run(() => {
+            expiresIn = json['expires_in'] || expiresIn;
+            refreshToken = json['refresh_token'] || refreshToken;
+            const expiresAt = this._absolutizeExpirationTime(expiresIn);
+            const data = assign(json, { 'expires_in': expiresIn, 'expires_at': expiresAt, 'refresh_token': refreshToken });
+            this._scheduleAccessTokenRefresh(expiresIn, null, refreshToken);
+            this.trigger('sessionDataUpdated', data);
+            resolve(data);
+          }).catch(reject);
+        }).catch(reject);
       }, (response) => {
         warn(`Access token could not be refreshed - server responded with ${response.responseJSON}.`, false, { id: 'ember-simple-auth.failedOAuth2TokenRefresh' });
```

Comment:
The same as https://github.com/simplabs/ember-simple-auth/pull/2017/files#r349651768, I think it doesn't make sense to have it here, this should probably go in https://github.com/simplabs/ember-simple-auth/pull/2017/files#diff-931dc51edd167948e773858f72a47233R442
I didn't find a test for this warning.

Reply:
see above